bool enable_reading
float32 percentage_threshold
bool instant